### PR TITLE
broker: refactor broker module loading code and fix minor bugs

### PR DESCRIPTION
**Pull request by garlick**
This was some broker cleanup that was the beginning of  work to create a "module loader" standalone helper executable that could be used to optionally load broker modules in a separate address space.

I didn't get that done, but this cleanup is progress on its own.

### CODE REVIEW

**Comment by garlick** [comment]
Restarted a builder that failed on an unrelated item:
```
2023-08-11T18:01:19.5529311Z ERROR: test_remote.t - missing test plan
2023-08-11T18:01:19.5530812Z ERROR: test_remote.t - exited with status 137 (terminated by signal 9?)
```

File: src/broker/broker.c [hunk 1]
```diff
@@ -1879,7 +1879,7 @@ static int module_insmod_respond (flux_t *h, module_t *p)
     int rc;
     int errnum = 0;
     int status = module_get_status (p);
-    flux_msg_t *msg = module_pop_insmod (p);
+    const flux_msg_t *msg = module_pop_insmod (p);
```
Comment (chu11) [inline src/broker/broker.c:1882]: commit message typo "but we struct" ... "but we use struct ..." or just "but struct ..."

**Review by chu11 — APPROVED** [review]
overall LGTM, just found a few nits

File: src/broker/modhash.h [hunk 2]
```diff
@@ -0,0 +1,66 @@
+/************************************************************\
+ * Copyright 2014 Lawrence Livermore National Security, LLC
```
[the remark this answers is not in the document]
Comment (garlick) [reply src/broker/modhash.h:2]: Except the code in this module is really from 2014...

File: src/broker/modhash.c [hunk 3]
```diff
@@ -0,0 +1,204 @@
+/************************************************************\
+ * Copyright 2014 Lawrence Livermore National Security, LLC
```
[the remark this answers is not in the document]
Comment (garlick) [reply src/broker/modhash.c:2]: Ibid.

**Comment by garlick** [comment]
Thanks @chu11 I fixed those items and re-pushed.  I left the (c) year 2014 since that code really is the old stuff.

**Review by chu11 — APPROVED** [review]

File: src/broker/broker.c [hunk 4]
```diff
@@ -480,10 +480,6 @@ int main (int argc, char *argv[])
         goto cleanup;
     }
 
-
-    /* Initialize module infrastructure.
-     */
-
```
Comment (chu11) [inline src/broker/broker.c]: oops, was this supposed to go into the commit prior to this one?

Comment (garlick) [reply src/broker/broker.c]: Aw dang it!  Good catch.  Will fix.

**Comment by garlick** [comment]
Tacked on a couple more commits that add testing.

**Review by grondo — APPROVED** [review]
I think I set this repo to dismiss approving reviews when a PR is updated. That is probably a premature setting at this point so I'll remove it for now.

However, I did just go through this and LGTM.

**Comment by garlick** [comment]
Thanks - I'll set MWP.

**Pull request merged**